plugins/services/src/js/components/modals/NewCreateServiceModalForm.js

```diff
@@ -2,81 +2,200 @@ import Ace from 'react-ace';
+import ReducerUtil from '../../../../../../src/js/utils/ReducerUtil';
+import ServiceFormSection from '../forms/ServiceFormSection';
 import TabButton from '../../../../../../src/js/components/TabButton';
 import TabButtonList from '../../../../../../src/js/components/TabButtonList';
-import TabView from '../../../../../../src/js/components/TabView';
+import Tabs from '../../../../../../src/js/components/Tabs';
 import TabViewList from '../../../../../../src/js/components/TabViewList';
+import Transaction from '../../../../../../src/js/structs/Transaction';
+import Util from '../../../../../../src/js/utils/Util';
 
-const METHODS_TO_BIND = ['handleJSONChange'];
+const METHODS_TO_BIND = [
+  'handleFormChange',
+  'handleFormBlur',
+  'handleJSONBlur',
+  'handleJSONChange'
+];
+
+const SECTIONS = [
+  ServiceFormSection
+];
 
 class NewCreateServiceModalForm extends React.Component {
   constructor() {
     super(...arguments);
 
+    let batch = new Batch();
+
+    let jsonParserReducers = ParserUtil.combineParsers(JSONParserReducers);
+    let jsonConfigReducers = ReducerUtil.combineReducers(JSONConfigReducers);
+    let inputConfigReducers = ReducerUtil.combineReducers(
+      Object.assign({}, ...SECTIONS.map((item) => {
+        return item.configReducers;
+      }))
+    );
+    let validationReducers = ReducerUtil.combineReducers(
+      Object.assign({}, ...SECTIONS.map((item) => {
+        return item.validationReducers;
+      }))
+    );
+
     this.state = {
-      jsonValue: ''
+      appConfig: {},
+      batch,
+      errors: {},
+      inputConfigReducers,
+      jsonConfigReducers,
+      jsonValue: JSON.stringify(batch.reduce(jsonConfigReducers, {}), null, 2),
+      jsonParserReducers,
+      validationReducers
     };
 
     METHODS_TO_BIND.forEach((method) => {
       this[method] = this[method].bind(this);
     });
   }
 
+  handleJSONBlur() {
+    try {
+      let {jsonParserReducers, jsonValue, validationReducers} = this.state;
+      let appConfig = JSON.parse(jsonValue);
+      let batch = new Batch();
+
+      // Run validation reducers on appConfig with new on minimal batch
+      let errors = Util.filterEmptyValues(
+        batch.reduce(validationReducers, appConfig)
+      );
+
+      // Add additional events after verification
+      jsonParserReducers(appConfig).forEach((item) => {
+        batch.add(item);
+      });
+
+      // This will essentially flush the batch
+      this.setState({appConfig: {}, batch, errors});
+    } catch (event) {
+      // TODO: handle error
+      let errors = Object.assign(
+        {},
+        this.state.errors,
+        {jsonEditor: 'JSON value is not valid json.'}
+      );
+      this.setState({errors});
+    }
+  }
+
   handleJSONChange(jsonValue) {
-    this.setState({jsonValue});
+    try {
+      let batch = new Batch();
+      let parsedData = JSON.parse(jsonValue);
+      let appConfig = batch.reduce(this.state.jsonConfigReducers, parsedData);
```

Comment:
Maybe we want to consider to do this outside of the try block because the try block will not be optimised.